### PR TITLE
feat: add pod topology spread constraints

### PR DESCRIPTION
**Pull request by ikatlinsky**
As per https://github.com/apache/apisix-helm-chart/issues/580

### CODE REVIEW

**Comment by ikatlinsky** [comment]
@tao12345666333 @lingsamuel @AlinsRan any chance we can merge this PR? Thanks!

**Review by tao12345666333 — APPROVED** [review]

File: charts/apisix-dashboard/values.yaml [hunk 1]
```diff
@@ -186,3 +186,7 @@ nodeSelector: {}
 tolerations: []
 
 affinity: {}
+
+# -- Topology Spread Constraints for pod assignment spread across your cluster among failure-domains
+# <br /> ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/#spread-constraints-for-pods
```
Comment (Gallardot) [inline charts/apisix-dashboard/values.yaml:191]: remove `<br />` ?

**Review by Gallardot — CHANGES_REQUESTED** [review]

File: charts/apisix/values.yaml [hunk 2]
```diff
@@ -155,6 +155,9 @@ apisix:
   tolerations: []
   # -- Set affinity for Apache APISIX deploy
   affinity: {}
+  # -- Topology Spread Constraints for pod assignment spread across your cluster among failure-domains
+  # <br /> ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/#spread-constraints-for-pods
```
Comment (Gallardot) [inline charts/apisix/values.yaml:159]: remove `<br />`

File: charts/apisix-ingress-controller/values.yaml [hunk 3]
```diff
@@ -166,6 +166,10 @@ updateStrategy: {}
 
 nodeSelector: {}
 tolerations: []
+affinity: {}
+# -- Topology Spread Constraints for pod assignment spread across your cluster among failure-domains
+# <br /> ref: https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/#spread-constraints-for-pods
```
Comment (Gallardot) [inline charts/apisix-ingress-controller/values.yaml:171]: remove `<br />`

**Review by Gallardot — CHANGES_REQUESTED** [review]

**Review by Gallardot — APPROVED** [review]

**Pull request merged**